experimental/desktop_photo_search/lib/main.dart

```diff
@@ -0,0 +1,123 @@
+// Copyright 2019 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:file_chooser/file_chooser.dart' as file_choser;
+import 'package:logging/logging.dart';
+import 'package:menubar/menubar.dart' as menubar;
+import 'package:meta/meta.dart';
+import 'package:provider/provider.dart';
+
+import 'src/model/photo_search_model.dart';
+import 'src/unsplash/unsplash.dart';
+import 'src/widgets/data_tree.dart';
+import 'src/widgets/photo_details.dart';
+import 'src/widgets/photo_search_dialog.dart';
+import 'src/widgets/split.dart';
+import 'unsplash_access_key.dart';
+
+void main() {
+  Logger.root.level = Level.ALL;
+  Logger.root.onRecord.listen((LogRecord rec) {
+    // ignore: avoid_print
+    print('${rec.loggerName} ${rec.level.name}: ${rec.time}: ${rec.message}');
+  });
+
+  if (unsplashAccessKey.isEmpty) {
+    Logger('main').severe('Unsplash Access Key is required. '
+        'Please add to `lib/unsplash_access_key.dart`.');
+    exit(1);
+  }
+
+  runApp(
+    ChangeNotifierProvider<PhotoSearchModel>(
+      create: (context) => PhotoSearchModel(
+        Unsplash(accessKey: unsplashAccessKey),
+      ),
+      child: UnsplashSearchApp(),
+    ),
+  );
+}
+
+class UnsplashSearchApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: 'Photo Search',
+      theme: ThemeData(
+        primarySwatch: Colors.orange,
+      ),
+      debugShowCheckedModeBanner: false,
+      home: const UnsplashHomePage(title: 'Photo Search'),
+    );
+  }
+}
+
+class UnsplashHomePage extends StatelessWidget {
+  const UnsplashHomePage({@required this.title});
+  final String title;
+
+  @override
+  Widget build(BuildContext context) {
+    final photoSearchModel = Provider.of<PhotoSearchModel>(context);
+    menubar.setApplicationMenu([
+      menubar.Submenu(label: 'Search', children: [
+        menubar.MenuItem(
+          label: 'Search ...',
+          onClicked: () {
+            showDialog(
+              context: context,
+              builder: (context) =>
+                  PhotoSearchDialog(photoSearchModel.addSearch),
+            );
+          },
+        ),
+      ])
+    ]);
+
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(title),
+      ),
+      body: photoSearchModel.entries.isNotEmpty
```

Comment:
Collection if doesn't have an else branch, so that replacement would result in collection if, followed by a collection if with the inverse logic, which breaks the natural this or that of the ternary expression.

Reply:
I'm kinda glad it isn't documented. It has the potential to lead to some really ungodly code if abused. Random else clauses just hanging out in a stack of widgets.